driverkit/utils/generate

```diff
@@ -48,8 +59,39 @@ ensure_kernelrelease() {
     TARGET_KERNEL_RELEASE="${TARGET_KERNEL%_${TARGET_KERNEL_VERSION}}"
 }
 
+function arch_to_driverkit_arch() {
+	case "$1" in
+		"x86_64")
+			echo -n "amd64"
+			;;
+		"aarch64")
+			echo -n "arm64"
+			;;
+		*)
+			echo "unknown architecture"
+			exit 1
+			;;
+	esac
+}
+
+# If driver version is a semver, we are in the newest case
+# thus we must support %arch/ subfolder for configs.
+# Else, we are in the old case, and we only support x86_64 configs
+# in the TARGET_VERSION folder.
+# TODO: rm this once old format support is dropped, and always use ${TARGET_VERSION}/${TARGET_ARCH}
+function path_from_version() {
+  rx='^([0-9]+\.){0,2}(\*|[0-9]+)$'
```

Comment:
We assume that new driverversions, that will use new directory tree, all have a semver.